lib/sensors/src/AEAT6012.cpp

```diff
@@ -0,0 +1,84 @@
+#include "AEAT6012.h"
+
+AEAT6012::AEAT6012::AEAT6012(PinName cs, PinName spi_mosi, PinName spi_clk, callback_ptr callback,
+                             uint32_t frequency_hz)
+    : m_position_raw(0), m_cs(cs), m_spi(NC, spi_mosi, spi_clk), m_callback(callback) {
+  m_spi.format(12, 2);
+  m_spi.frequency(frequency_hz);
+  m_spi.set_dma_usage(DMA_USAGE_ALWAYS);
+
+  m_cs = 1;  // Keep CS high until a read is requested
+}
+
+AEAT6012::AEAT6012::AEAT6012(const Config &config)
+    : AEAT6012(config.cs, config.spi_mosi, config.spi_clk, config.callback, config.frequency_hz) {}
+
+float AEAT6012::AEAT6012::read_position(void) {
+  // Assert CS to start signaling
+  m_cs = 0;
```

Comment:
dont pass in call back function ptr in constructor pass it in when u call the read function. for example
// async version of read
bool readAngle(float &angle, callback_ptr cb);

make sure u follow the pattern layed out in the encoder rearchitecture

Reply:
freq shouldnt be a varaible. its an internal detail we shouldnt be exposing outside. it should be a private static constexpr